### ISSUE TITLE
Support Sunrise PT-PU (platform ptpu)

### ISSUE THREAD

**Issue by tengqm** [flagos-ai/Megatron-LM-FL#183]
### What is missing

Megatron-LM-FL has no platform for Sunrise PT-PU. `torch_ptpu` registers its
backend through `rename_privateuse1_backend` and does not alias `torch.cuda`, so
on a PPU host none of the existing vendor checks in `platform_manager` fires and
the unconditional `cpu` platform is selected — at which point the training entry
point aborts with "Megatron requires an accelerator."

### What a run needs beyond the platform wrapper

1. **The device's operator library has to be bound.** `aten::tanh.out` is
   reached from the jit-fuser warmup in `initialize.py`, before any argument can
   turn that path off, and the native implementation does not cover it.
2. **The backend's collectives have to be selectable.** PCCL registers under its
   own name and is not an alias of nccl, so `--distributed-backend` has to
   accept it the way it already accepts `mccl`.
3. **Gradient/parameter collectives must not require the `*_coalesced` entry
   points.** PCCL implements `all_reduce`, `broadcast`, `all_gather` /
   `all_gather_into_tensor`, `reduce_scatter` (list form) and `barrier`, but not
   `allreduce_coalesced` / `allgather_into_tensor_coalesced` /
   `reduce_scatter_tensor_coalesced` (`reduce_scatter_tensor` and
   `all_to_all_single` are missing too, and `Backend.supports_coalescing` is
   False). Megatron's parameter all-gather and gradient all-reduce went straight
   to `torch.distributed._coalescing_manager`, which needs all three.

### Proposal

#181 adds the platform, binds the operator library at registration, selects
`ptpu` before `cpu`, and replays recorded collectives one at a time when the
backend reports no coalescing support — the shape torch itself uses in
`batch_isend_irecv`. Verified with the standard mock-data `pretrain_gpt`
recipe on a PPU node:

```
Megatron-LM-FL Platform: ptpu Selected
iteration 1/5 | lm loss: 1.089504E+01 | grad norm: 1.867
validation loss at iteration 5 | lm loss value: 1.088120E+01
exit 0
```

With `--distributed-backend pccl`, as the recipe notes in the PR.
